src/crsAdapter/BewotecExpertAdapter.js

```diff
@@ -306,11 +306,17 @@ class BewotecExpertAdapter {
                     return;
                 }
 
+                this.logger.info('received data from bewotec data bridge: ');
+
+                if (!this.options.debug && this.bridgeWindow && !this.bridgeWindow.closed) {
+                    this.logger.info('bewotec data bridge will be closed now');
+                    this.bridgeWindow.close();
+                }
+
                 if (this.helper.window.removeEventListener) {
                     this.helper.window.removeEventListener('message', bewotecDataListener);
                 }
 
-                this.logger.info('received data from bewotec data bridge: ');
                 this.logger.info(message.data);
```

Comment:
this line is hard connected to the line you moved a few lines to the top. you should move that too.